app/assets/javascripts/shopify_app/storage_access_redirect.js

```diff
@@ -0,0 +1,17 @@
+(function() {
+  function redirect() {
+    var redirectTargetElement = document.getElementById("redirection-target");
+
+    var targetInfo = JSON.parse(redirectTargetElement.dataset.target)
+
+    if (window.top == window.self) {
+      // If the current window is the 'parent', change the URL by setting location.href
+      window.top.location.href = targetInfo.hasStorageAccessUrl;
+    } else {
+        var storageAccessHelper = new StorageAccessHelper(targetInfo);
```

Comment:
I think we can simplify this even further. Both times StorageAccessHelper is constructed `.execute()` is called on it right after. The only difference is the additional logic above that instead redirects immediately if called as the parent frame. Why don't we move this logic in the access helper itself, which could be moved into the helper itself and allow us to test it. I think that along with my [other comment above](https://github.com/Shopify/shopify_app/pull/647/files#r229396355) would allow us to get rid of this file and `partition_cookies.js` and save a couple http requests.